rig-core/src/completion.rs

```diff
@@ -432,3 +472,78 @@ impl<M: CompletionModel> CompletionRequestBuilder<M> {
         model.completion(self.build()).await
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn test_document_display_without_metadata() {
+        let doc = Document {
+            id: "123".to_string(),
+            text: "This is a test document.".to_string(),
+            additional_props: HashMap::new(),
+        };
+
+        let expected = "<file id: 123>\nThis is a test document.\n</file>\n";
+        assert_eq!(format!("{}", doc), expected);
+    }
+
+    #[test]
+    fn test_document_display_with_metadata() {
+        let mut additional_props = HashMap::new();
+        additional_props.insert("author".to_string(), "John Doe".to_string());
+        additional_props.insert("length".to_string(), "42".to_string());
+
+        let doc = Document {
+            id: "123".to_string(),
+            text: "This is a test document.".to_string(),
+            additional_props,
+        };
+
+        let expected = concat!(
+            "<file id: 123>\n",
+            "<metadata author: \"John Doe\" length: \"42\" />\n",
+            "This is a test document.\n",
+            "</file>\n"
+        );
+        assert_eq!(format!("{}", doc), expected);
+    }
```

Comment:
This test is not deterministic since the ordering of KVs in `additional_props` is not deterministic. 
Either change the test to compare the two scenarios, or change the `Display` implementation to get a deterministic output (e.g.: KVs ordered alphabetically)